src/kokkos_ext/ArborX_DetailsKokkosExtScopedProfileRegion.hpp

```diff
@@ -12,6 +12,15 @@
 #ifndef ARBORX_DETAILS_KOKKOS_EXT_SCOPED_PROFILE_REGION_HPP
 #define ARBORX_DETAILS_KOKKOS_EXT_SCOPED_PROFILE_REGION_HPP
 
+#include <Kokkos_Macros.hpp>
+
+#if KOKKOS_VERSION >= 40099
+#include <Kokkos_Profiling_ScopedRegion.hpp>
```

Comment:
I would have guarded with `#if __has_include(Kokkos_Profiling_ScopedRegion.hpp)` and just commented that it is provided in 4.1 but what you did is good enough